packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -2924,14 +2919,12 @@ describe('KeyringController', () => {
         it('should unlock the wallet also if encryption parameters are outdated and the vault upgrade fails', async () => {
           await withController(
             {
-              skipVaultCreation: true,
```

Comment:
Initial vault creation was being skipped in `submitPassword` tests because the default `withController` behavior is to generate a vault and leave the controller unlocked. So when these tests get executed KeyringController will already be unlocked. 

For some of these test cases, it shouldn't be a problem. Though for some of them, there's the risk of getting false positives

Reply:
initially i've changed it because the `withController` function didn't allow to setup with encryption key. i've changed that now, so will check if i can change it back in some cases. would be good to know in which cases it's problematic to have this skipped.

Reply:
I think it's potentially problematic for any test case that aims to verify how unlocking works, since the controller is already unlocked _before_ the tests.
e.g. if a test case asserts `expect(controller.state.isUnlocked).toBe(true);`, that would potentially be a false positive

Reply:
> tbh, I think it might be better to simply lock the controller in cases where you need a locked controller.

This may work in some cases, though you'd have to use another public function to test another one, which is ok, but better to avoid it if we can make the test simpler and more independent

> the issue i had with setting up the vault like this is that it depends on the implementation of the mock controller

Perhaps you are referring to the encryptor? If so, that's the reason why we set up these mock responses:

```typescript
jest.spyOn(encryptor, 'decryptWithKey').mockResolvedValueOnce([
  {
    type: KeyringTypes.hd,
    data: '0x123',
  }
);
```

Reply:
Sorry. Yes. Mock encryptor